boot/boot.go

```diff
@@ -442,6 +451,37 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
 		return fmt.Errorf("cannot set recovery system environment: %v", err)
 	}
 
+	// get the ubuntu-boot grub and extrac the kernel there
+	opts = &bootloader.Options{
+		// TODO:UC20: we use "recovery: true" here because on
+		// the partition the file layout of ubuntu-boot looks
+		// the same as ubuntu-seed
+		Recovery:              true,
+		ForceKernelExtraction: true,
+	}
+	bl, err = bootloader.Find(filepath.Join(runMnt, "ubuntu-boot"), opts)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot find run system bootloader: %v", err)
+	}
+	kernelf, err := snap.Open(bootWith.KernelPath)
+	if err != nil {
+		return err
+	}
+	// TODO:UC20: extract in the new "static" image way
+	if err := bl.ExtractKernelAssets(bootWith.Kernel, kernelf); err != nil {
+		return err
+	}
```

Comment:
we need also to copy kernel and base to ubuntu-data/var/lib/snapd/snaps

Reply:
Thanks! I added a (hopefully) uncontroversial PR for this in https://github.com/snapcore/snapd/pull/7918